### PR TITLE
Update XML docs for public Monitor Query APIs

### PR DESCRIPTION
**Pull request by scottaddie**
Addresses https://github.com/Azure/azure-sdk-for-net/issues/37933 by linking customers to the mocking doc on learn.microsoft.com.

Addresses https://github.com/Azure/azure-sdk-for-net/issues/38707 by explaining the relationship between the `query` and `timeRange` parameters.

### CODE REVIEW

**Review by nisha-bhatia — APPROVED** [review]

**Comment by nisha-bhatia** [comment]
Just curious why you renamed timeSpan to timeRange? :) is it to make it more clear?

**Comment by scottaddie** [comment]
> Just curious why you renamed timeSpan to timeRange? :) is it to make it more clear?

Good question. The camel casing of `timeSpan` led me to believe this was a reference to the parameter, which is named `timeRange`. Note the casing of "timespan" earlier in the description, which was another clue. Does that make sense, or did I misunderstand?

File: sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs [hunk 1]
```diff
@@ -165,12 +171,15 @@ public virtual async Task<Response<IReadOnlyList<T>>> QueryWorkspaceAsync<T>(str
         /// <summary>
         /// Executes the logs query.
         /// </summary>
-        /// <param name="workspaceId">The workspace id to include in the query (<c>xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx</c>).</param>
-        /// <param name="query">The Kusto query to execute.</param>
-        /// <param name="timeRange">The timespan over which to query data. Logs will be filtered to include entries produced starting at <c>Now - timeSpan</c>. </param>
+        /// <param name="workspaceId">The workspace ID to include in the query (<c>xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx</c>).</param>
+        /// <param name="query">TThe Kusto query to fetch the logs.</param>
```
Comment (nisha-bhatia) [inline sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs:175]: nit: spelling of The

File: sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs [hunk 2]
```diff
@@ -189,12 +198,15 @@ public virtual Response<LogsQueryResult> QueryWorkspace(string workspaceId, stri
         /// <summary>
         /// Executes the logs query.
         /// </summary>
-        /// <param name="workspaceId">The workspace id to include in the query (<c>xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx</c>).</param>
-        /// <param name="query">The Kusto query to execute.</param>
-        /// <param name="timeRange">The timespan over which to query data. Logs will be filtered to include entries produced starting at <c>Now - timeSpan</c>. </param>
+        /// <param name="workspaceId">The workspace ID to include in the query (<c>xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx</c>).</param>
+        /// <param name="query">The Kusto query to fetch the logs.</param>
+        /// <param name="timeRange">The time period for which the logs should be looked up.</param>
```
Comment (nisha-bhatia) [inline sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs:203]: I think this is good now!

**Pull request merged**